### ISSUE TITLE
Unable to train model with Tensorflow Object Detection API

### ISSUE THREAD

**Issue by dnlanoj** [ROCmSoftwarePlatform/tensorflow-upstream#1085]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): Doker images : rocm/tensorflow:rocm3.5-tf1.15
- TensorFlow version (use command below): 1.15.0
- Python version: 3.6.9
-Rocm version : 3.5 
- GPU model and memory: Radeon RX580

**Describe the current behavior**

I am trying to train a custom Mask RCNN using tensorflow object detection API. I was previously using Tensorflow 2.2.0 with rocm 3.5 (from docherhub) and was able to train a Mask RCNN using this repositery (even though I remembered having a hard time during the installation process already) : https://github.com/matterport/Mask_RCNN. I then had to go to tensorflow 1.15.0  and use the object detection api due to model compatibility and it is now impossible to train the network. 
When I launch the training process all the library are opened correctly, the GPU is recognised, VRAM full and GPU% alternates between full capacity and 0 however the fans do not go higher than 40% (although the temperature does not really requires it). Then several issues usually happen, sometimes nothing happens for a long time, no GPU usage, the CPU's going quite crazy and when quitting the application with ^C it prompts "Unable to find a suitable algorithm for doing backward filter convolution", sometimes it takes much longer, a few steps are done and I have an weird issue like : 
              " (0) Invalid argument: assertion failed: [maximum box coordinate value is larger than 1.100000: ] [5597.49268]
	              [[{{node Loss/BoxClassifierLoss/ToAbsoluteCoordinates/Assert/AssertGuard/Assert}}]]
	              [[ToAbsoluteCoordinates/Assert/AssertGuard/Assert/data_0/_6548]]
                 (1) Invalid argument: assertion failed: [maximum box coordinate value is larger than 1.100000: ] [5597.49268]
	               [[{{node Loss/BoxClassifierLoss/ToAbsoluteCoordinates/Assert/AssertGuard/Assert}}]]
                 0 successful operations.
                 0 derived errors ignored."
In any case the Losses are always nonsense and no predictions are made even if I'm quite sure that the pipeline is quite okay.
Although I noticed that the memory seems to be limited as I can't go higher than a batch size of 1 or I have OOM issues even if I am working on images of size 480*480.

I can also say that whenever the training is started it is completely impossible to use the computer as all resources seem to be dedicated to the training.

I am however able to train a simple sequential model on the MNIST database with a respectable 32 us/step but when the model gets a bit more complicated like CIFAR model classifier, I have again other issues (example of issue : No registered '_FusedConv2D' OpKernel for GPU devices compatible with node {{node conv_4_features/activation_5/Relu}} )

**Standalone code to reproduce the issue**
I'm just doing : 
root@**********:/tf1_docker_share# python3 object_detection/model_main.py --pipeline_config_path=../../MASK_RCNN_InceptionV2/models/my_mrcnn_dir/mrcnn.config --model_dir=../../MASK_RCNN_InceptionV2/models/my_mrcnn_dir/saved_model/ --num_train_steps=200 --sample_1_of_n_eval_samples=1 --alsoloftostderr

**Other info / logs**
in the attachments.

Please could you help me point me to the configuration I could try in order to use my AMD GPU? 
I can give more info if needed,
thanks a lot.

**Comment by dnlanoj** [comment]
Hi,
for people who might be struggling as well I have downgraded to tensorflow 1.14.0 and the training is a bit more stable, however it is still very long (one hour per step) and gives no concluants results. 
It also takes a lot of GPU RAM so that after a few steps (so a few hours) the process is killed without further indications.
Hope you can help me.

**Comment by jerryyin** [comment]
Please note TF version and ROCm version needs to be aligned if you try with a past branch. Please see [here](https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/blob/develop-upstream/rocm_docs/tensorflow-rocm-release.md).

Performance wise, I suggest to use the latest ROCm version in [this channel](https://hub.docker.com/repository/registry-1.docker.io/rocm/tensorflow/tags?page=1).

**Comment by dnlanoj** [comment]
Hello thank you for your answer ! However as I said I used the docker images from dockerhub using the tensorflow 1.14 version so I am now with tf 1.14.6 and rocm 3.1 which is aligned regarding the compability table and I still have my problems : around one hour per step and process get killed after several steps due to the RAM I suppose (I have 32 Gb of RAM).

**Comment by jerryyin** [comment]
Sorry if I didn't make it clear. We only support the latest `1.x` TensorFlow or `2.x` TensorFlow also with latest ROCm version. 1.14.6 was released half a year ago and we have fixed numerous issues like yours in the time range.

**Comment by dnlanoj** [comment]
Okay thank you for your answer ! I'm closing this issue, I didn't manage to make it work with tf 1.15 but it worked with the docker image tf 1.14.6 and using the legacy/train.py from the tensorflow object detection api instead of the model_main.py.
